tests/test_splitting.py

```diff
@@ -287,6 +288,12 @@ def test_split_indices():
     assert samples_left.shape[0] == si_root.n_samples_left
     assert samples_right.shape[0] == si_root.n_samples_right
 
+    samples_left_single_thread, samples_right_single_thread = split_indices_single_thread(
```

Comment:
Maybe instead parametrize this test with `parallel_split=True` and `parallel_split=False`, so that both methods go through exactly the same tests (though I agree your current test also covers this implicitly)